<a href="https://colab.research.google.com/github/Luz-Dobbins/Project-2/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
import pandas as pd

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


#First choice: dataset 1

In [6]:
filename = '/content/drive/MyDrive/Data/Metabolic  Syndrome.csv'
ms_df = pd.read_csv(filename)
ms_df.head()

seqn  Age     Sex  Marital  Income   Race  WaistCirc   BMI  Albuminuria  \
0  62161   22    Male   Single  8200.0  White       81.0  23.3            0   
1  62164   44  Female  Married  4500.0  White       80.1  23.2            0   
2  62169   21    Male   Single   800.0  Asian       69.6  20.1            0   
3  62172   43  Female   Single  2000.0  Black      120.4  33.3            0   
4  62177   51    Male  Married     NaN  Asian       81.1  20.1            0   

   UrAlbCr  UricAcid  BloodGlucose  HDL  Triglycerides MetabolicSyndrome  
0     3.88       4.9            92   41             84         No MetSyn  
1     8.55       4.5            82   28             56         No MetSyn  
2     5.07       5.4           107   43             78         No MetSyn  
3     5.22       5.0           104   73            141         No MetSyn  
4     8.13       5.0            95   43            126         No MetSyn

In [7]:
ms_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2401 entries, 0 to 2400
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   seqn               2401 non-null   int64  
 1   Age                2401 non-null   int64  
 2   Sex                2401 non-null   object 
 3   Marital            2193 non-null   object 
 4   Income             2284 non-null   float64
 5   Race               2401 non-null   object 
 6   WaistCirc          2316 non-null   float64
 7   BMI                2375 non-null   float64
 8   Albuminuria        2401 non-null   int64  
 9   UrAlbCr            2401 non-null   float64
 10  UricAcid           2401 non-null   float64
 11  BloodGlucose       2401 non-null   int64  
 12  HDL                2401 non-null   int64  
 13  Triglycerides      2401 non-null   int64  
 14  MetabolicSyndrome  2401 non-null   object 
dtypes: float64(5), int64(6), object(4)
memory usage: 281.5+ KB


1. Source of data

The data.world site from the preapproved list. https://data.world/informatics-edu/metabolic-syndrome-prediction

2. Brief description of data

This dataset will help me access the proprability that someone will get Metabolic Syndrome. Which could lead to other health conditions.

3. What is the target?

The target is Metabolic Syndrome

4. What does one row represent? (A person?  A business?  An event? A product?)

Each row represents a person who does or does not have the syndrome.

5. Is this a classification or regression problem?

This is a classifier because it's a yes or no question.

6. How many features does the data have?

The data has 13 features. I am not counting the seqn column it is not needed for the prediction.

7. How many rows are in the dataset?

There are 2401 rows.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

I see that ther is missingt data. I also need to encode the variables and scale the data. I plan on using classifier modeling , tuning of metric models(accuracy or recall) and GridSearch.


#Second choice: dataset 2

In [8]:
path = '/content/drive/MyDrive/Data/healthcare-dataset-stroke-data.csv'
s_df = pd.read_csv(path)
s_df.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [9]:
s_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


1. Source of data

I selected this dataset from the approved list. https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset?resource=download

2. Brief description of data

This dataset will access me in predicting if a person will have a stroke based on this specific perameters.

3. What is the target?

The target will be stroke

4. What does one row represent? (A person?  A business?  An event? A product?)

The rows represent a person.

5. Is this a classification or regression problem?

This is a classification because its a yes or no problem.

6. How many features does the data have?

There are 10 featuyres. I am going to remove the id column, it is not needed for this prediction.

7. How many rows are in the dataset?

There are 5110 rows in this dataset.

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

I have missing data to fix and will be imputing objects and scaling the numerical data. I plan on using classifier modeling , tuning of metric models(accuracy or recall) and GridSearch.